src/AWS.Deploy.CLI/Commands/GenerateDeploymentProjectCommand.cs

```diff
@@ -0,0 +1,185 @@
+        {
+            var recommendations = await orchestrator.GenerateDeploymentRecommendations();
+            if (recommendations.Count == 0)
+            {
+                throw new FailedToGenerateAnyRecommendations("The project you are trying to deploy is currently not supported.");
+            }
+            return recommendations;
+        }
+
+        private string GenerateDefaultSaveDirectory(ProjectDefinition projectDefinition)
+        {
+            var projectDirectoryFullPath = new DirectoryInfo(projectDefinition.ProjectPath).Parent.FullName;
+            var saveDirectory = projectDirectoryFullPath + "DeploymentProject";
+
+            var suffixNumber = 0;
+            while (Directory.Exists(saveDirectory))
+                saveDirectory = projectDirectoryFullPath + $"DeploymentProject{++suffixNumber}";
+            
+            return saveDirectory;
+        }
+
+        private bool ValidateSaveDirectory(string saveDirectory)
+        {
+            var directoryCreatedByDeployTool = false;
+
+            if (Directory.Exists(saveDirectory))
+            {
+                CheckDirectoryEmpty(saveDirectory);
+            }
+            else
+            {
+                CreateNewDirectory(saveDirectory);
+                directoryCreatedByDeployTool = true;
+            }
+
+            CheckSaveDirectoryInsideProjectDirectory(saveDirectory, _session.ProjectDefinition, directoryCreatedByDeployTool);
+
+            return directoryCreatedByDeployTool;
+        }
+
+        private void CheckDirectoryEmpty(string saveDirectory)
+        {
+            if (Directory.GetFiles(saveDirectory).Length > 0 || Directory.GetDirectories(saveDirectory).Length > 0)
+            {
+                var errorMessage = "The directory specified for saving the CDK project is non-empty. " +
+                    "Please provide an empty directory path and try again.";
+                throw new InvalidSaveDirectoryForCdkProject(errorMessage);
+            }
+        }
+            
+
+        private void CreateNewDirectory(string saveDirectory)
+        {
+            try
+            {
+                Directory.CreateDirectory(saveDirectory);
+            }
+            catch (Exception e)
+            {
+                var errorMessage = "Failed to create a directory at the specified path as it contains one or more invalid characters";
+                throw new InvalidSaveDirectoryForCdkProject(errorMessage, e);
+            }
+        }
+
+        private void CheckSaveDirectoryInsideProjectDirectory(string saveDirectory,
+            ProjectDefinition projectDefinition, bool directoryCreatedByDeployTool)
+        {
+            var saveDirectoryFullPath = new DirectoryInfo(saveDirectory).FullName;
+            var projectDirectoryFullPath = new DirectoryInfo(projectDefinition.ProjectPath).Parent.FullName;
+
+            if (saveDirectoryFullPath.Contains(projectDirectoryFullPath + "\\", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
This is cross platform code, we should use `Path.Join` or `Path.DirectorySeparatorChar`.

Reply:
Fixed